targets/csharp/source/source/Extensions/Models/PlayFabCloudScriptModelsExtensions.cs.ejs

```diff
@@ -0,0 +1,61 @@
+﻿namespace PlayFab.CloudScriptModels
+{
+    public class FunctionExecutionContext<T>
+    {
+        public PlayFabApiSettings ApiSettings { get; private set; }
+
+        public ProfilesModels.EntityProfileBody CallerEntityProfile { get; set; }
+
+        public T FunctionArgument { get; set; }
+
+        protected FunctionExecutionContext() { }
+
+        public static async System.Threading.Tasks.Task<FunctionExecutionContext<T>> Create(System.Net.Http.HttpRequestMessage request)
+        {
+            string body = await request.Content.ReadAsStringAsync();
+            var contextInternal = Json.PlayFabSimpleJson.DeserializeObject<FunctionExecutionContextInternal>(body);
+            var settings = new PlayFabApiSettings
+            {
+                TitleId = contextInternal.TitleAuthenticationContext.Id,
+                DeveloperSecretKey = contextInternal.TitleAuthenticationContext.SecretKey
+            };
+
+            // TODO: STATIC ENTITY TOKEN?
+            PlayFabSettings.staticPlayer.EntityToken = contextInternal.TitleAuthenticationContext.EntityToken;
+            return new FunctionExecutionContext<T>()
+            {
+                ApiSettings = settings,
+                CallerEntityProfile = contextInternal.CallerEntityProfile,
+                FunctionArgument = (T)contextInternal.FunctionArgument
+            };
+        }
+
+        private class FunctionExecutionContextInternal<V>
```

Comment:
Why did you use 'V' here? It's always going to be 'T' anyway.

Reply:
Warning	CS0693	Type parameter 'T' has the same name as the type parameter from outer type

Due to the above warning that is thrown when you create a generic type nested within another generic type.